src/server/game/DataStores/DB2Structure.h

```diff
@@ -2877,6 +2942,56 @@ struct MawPowerEntry
     int32 MawPowerRarityID;
 };
 
+struct MCRSlotXMCRCategoryEntry
+{
+    int32 ID;
+    int32 ModifiedCraftingCategoryID;
+    int32 Order;
+    int32 ModifiedCraftingReagentSlotID;
+};
+
+struct ModifiedCraftingCategoryEntry
+{
+    int32 ID;
+    LocalizedString DisplayName;
+    LocalizedString Description;
+    int32 Field_9_0_1_33978_001;
+    float MatQualityWeight;
+    int32 Field_10_0_0_44649_004;
+};
+
+struct ModifiedCraftingReagentItemEntry
+{
+    int32 ID;
+    LocalizedString Description;
+    int32 ModifiedCraftingCategoryID;
+    int32 ItemBonusTreeID;
+    int32 Flags;
+    int32 Field_9_1_0_38511_004;
+    int32 ItemContextOffset;
+};
+
+struct ModifiedCraftingReagentSlotEntry
+{
+    LocalizedString Name;
+    int32 ID;
+    int32 Field_9_0_1_33978_001;
+    int32 PlayerConditionID;
+    int32 ReagentType;
+    int8 Field_10_0_2_46091_005;
+};
+
+struct ModifiedCraftingSpellSlotEntry
+{
+    int32 ID;
```

Comment:
not uint? 👀

Reply:
there is no need to play smartass in db2 structs because the field types will be checked against the client extracted db2 meta on startup. If it loads it's correct. If not, he'll fix it. See the WIP in the title.

Reply:
i don't know what the comment about the WIP or about playing smartass have to do with anything I pointed out, there's no reason not to be using unsigned integers when the ID field only stores positive values

Reply:
In any case, the whole crafting system is huge and it's best to break it down to multiple PRs, so it's easier to review. Good so far!

Reply:
It should indeed be uint32, simply follow the rest of the structures.